### PR TITLE
Add com.jwestall.Forecast

### PR DESCRIPTION
**Pull request by jwestall**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  A simple weather application written in rust and libcosmic.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission.

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: com.jwestall.Forecast.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": ["org.freedesktop.Sdk.Extension.rust-stable"],
+    "command": "cosmic-ext-forecast",
+    "finish-args": [
+      "--share=ipc",
+      "--socket=fallback-x11",
+      "--socket=wayland",
+      "--device=dri",
+      "--talk-name=com.system76.CosmicSettingsDaemon",
+      "--filesystem=xdg-config/cosmic:ro"
```
Comment (hfiguiere) [inline com.jwestall.Forecast.json:14]: why do you need this?

File: com.jwestall.Forecast.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    ],
+    "build-options": {
+      "append-path": "/usr/lib/sdk/rust-stable/bin",
+      "build-args": ["--share=network"],
```
Comment (hfiguiere) [inline com.jwestall.Forecast.json:18]: can't use this. Build is sandbox and must be offline.

File: com.jwestall.Forecast.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      {
+        "name": "just",
+        "buildsystem": "simple",
+        "build-commands": ["install -Dm0755 just /app/bin/just"],
+        "sources": [
+          {
+            "type": "archive",
+            "strip-components": 0,
+            "dest-filename": "just.tar.gz",
+            "url": "https://github.com/casey/just/releases/download/1.25.2/just-1.25.2-x86_64-unknown-linux-musl.tar.gz",
```
Comment (hfiguiere) [inline com.jwestall.Forecast.json:33]: not sure why you download binaries but this runtim surely doesn't use musl.

File: com.jwestall.Forecast.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        ]
+      },
+      {
+        "name": "cosmic-ext-forecast",
+        "buildsystem": "simple",
+        "build-commands": ["just vendor", "just build-vendored", "just flatpak"],
+        "sources": [
+          {
+            "type": "git",
+            "branch": "main",
```
Comment (hfiguiere) [inline com.jwestall.Forecast.json:54]: never a git branch.

File: com.jwestall.Forecast.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            "url": "https://github.com/casey/just/releases/download/1.25.2/just-1.25.2-aarch64-unknown-linux-musl.tar.gz",
+            "sha256": "049d624255d150f18b0f3a1257a163366b0dc22e16526214773903125145e515",
+            "only_arches": ["aarch64"]
+          }
+        ]
+      },
+      {
+        "name": "cosmic-ext-forecast",
+        "buildsystem": "simple",
+        "build-commands": ["just vendor", "just build-vendored", "just flatpak"],
```
Comment (hfiguiere) [inline com.jwestall.Forecast.json:50]: please format the JSON so it is readable.

File: com.jwestall.Forecast.json (hunk 3 above)
Comment (hfiguiere) [reply com.jwestall.Forecast.json:33]: also if this is build only then the `cleanup` rule is missing to remove i.

File: com.jwestall.Forecast.json [hunk 6]
```diff
@@ -0,0 +1,60 @@
+{
+    "id": "com.jwestall.Forecast",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "23.08",
```
Comment (hfiguiere) [inline com.jwestall.Forecast.json:4]: current version is 24.08

File: com.jwestall.Forecast.json (hunk 1 above)
Comment (jwestall) [reply com.jwestall.Forecast.json:14]: I thought they were needed to interact with the cosmic settings daemon. I will remove them when I can get back to my computer.

File: com.jwestall.Forecast.json (hunk 2 above)
Comment (jwestall) [reply com.jwestall.Forecast.json:18]: Noted, will remove.

File: com.jwestall.Forecast.json (hunk 1 above)
Comment (hfiguiere) [reply com.jwestall.Forecast.json:14]: "they". It's only line 14 the question.

Comment (jwestall) [reply com.jwestall.Forecast.json:14]: So line 14 is needed in order for the app to sync with the system theme on the Cosmic desktop. See [Eduado's blog post](https://hachyderm.io/@edfloreshz/112769206427407768) for more details.

Comment (bbhtt) [reply com.jwestall.Forecast.json:14]: You have to ask an exception for this https://docs.flathub.org/docs/for-app-authors/linter#exceptions.

Comment (bbhtt) [reply com.jwestall.Forecast.json:14]: if you run the linter as described in the submission docs, you can get the error code

File: com.jwestall.Forecast.json [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+      },
+      {
+        "name": "cosmic-ext-forecast",
+        "buildsystem": "simple",
+        "build-commands": ["just vendor", "just build-vendored", "just flatpak"],
+        "sources": [
+          {
+            "type": "git",
+            "branch": "main",
+            "url": "https://github.com/cosmic-utils/forecast.git"
```
Comment (bbhtt) [inline com.jwestall.Forecast.json:55]: why is the repo needed if you are already installing binaries ?

File: com.jwestall.Forecast.json [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            "install -Dm0644 ./res/com.jwestall.Forecast.metainfo.xml /app/share/metainfo/com.jwestall.Forecast.metainfo.xml",
+            "install -Dm0644 ./res/icons/hicolor/scalable/apps/com.jwestall.Forecast.svg /app/share/icons/hicolor/scalable/apps/com.jwestall.Forecast.svg"
+        ],
+        "sources": [
+          {
+            "type": "git",
+            "url": "https://github.com/cosmic-utils/forecast.git",
+            "commit": "bb5e4cd75f602475c04c899bd5da12c6d020d10c"
+          },
+	  "cargo-sources.json"
```
Comment (bbhtt) [inline com.jwestall.Forecast.json:43]: can you indent this properly ?

Comment (jwestall) [reply com.jwestall.Forecast.json:43]: Would line 43 not be in the same indentation as line 42? (I'll be happy to fix indentation just looking for clarification)

Comment (bbhtt) [reply com.jwestall.Forecast.json:43]: No it should start at the `},`, that whole file is another source.

Comment (jwestall) [reply com.jwestall.Forecast.json]: I see it now, should be fixed per latest commit.

**Comment by bbhtt** [comment]
bot, build com.jwestall.Forecast

**Comment by bbhtt** [comment]
Remove cosmic from categories and desktop file, it was added last month to xdg-specs and no release has been made since then, so it is failing validation.

A  different tag for marking apps suited for COSMIC see the discussion https://github.com/mariinkys/starrydex/pull/3#issuecomment-2380880405


You are also missing releases https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#release

**Comment by bbhtt** [comment]
I will update desktop-file-utils to the latest master commit. So you can just fix the metainfo file - remove cosmic from category and add releases.

**Comment by jwestall** [comment]
I updated the desktop file and metainfo file from master branch as requested.

**Comment by bbhtt** [comment]
bot, build com.jwestall.Forecast

**Comment by bbhtt** [comment]
Missing OARS tags https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#oars-information

**Comment by jwestall** [comment]
Tag added to master branch

**Comment by bbhtt** [comment]
bot, build com.jwestall.Forecast

**Comment by ahoneybun** [comment]
That testing repo works for me.

screenshot-2024-09-30-16-19-31

**Comment by bbhtt** [comment]
Exit buttons are invisible and a shadow shows up on hover. Is that intended ?


https://github.com/user-attachments/assets/a86a8577-04fd-41b0-ae12-4bc60e6ae6ce

**Comment by jwestall** [comment]
I don't believe that is how it should run. I am reaching out to S76 to see if that's a bug in the toolkit. It was developed and intended to run on the COSMIC desktop.

**Comment by bbhtt** [comment]
Being discussed at https://github.com/flathub/flathub/pull/5649#issuecomment-2385069162

**Comment by jwestall** [comment]
It was also discussed [here](https://github.com/pop-os/libcosmic/issues/550) and a fix was found for the cosmic web app Manager [here](https://github.com/cosmic-utils/web-apps/blob/master/io.github.elevenhsoft.WebApps.json#L55-L70).

I can update my manifest as well if that solution is acceptable.

**Comment by bbhtt** [comment]
> and a fix was found for the cosmic web app Manager [here](https://github.com/cosmic-utils/web-apps/blob/master/io.github.elevenhsoft.WebApps.json#L55-L70).

Yes bundling it is fine.

**Comment by bbhtt** [comment]
bot, build com.jwestall.Forecast

**Comment by bbhtt** [comment]
You have to copy the just module above as well, and add a `cleanup: ["*"]` to it as it's only required at build time.

**Comment by bbhtt** [comment]
bot, build com.jwestall.Forecast

**Pull request closed without merging**